src/components/camera-tool.js

```diff
@@ -100,12 +106,38 @@ AFRAME.registerComponent("camera-tool", {
     this.el.removeEventListener("stateadded", this.stateAdded);
   },
 
+  remove() {
+    if (this.cameraSystem) {
+      this.cameraSystem.deregister(this.el);
+    }
+  },
+
   stateAdded(evt) {
     if (evt.detail === "activated") {
       this.takeSnapshotNextTick = true;
     }
   },
 
+  focus(el, track) {
+    this.lookAt(el);
+
+    if (track) {
+      this.trackTarget = el;
+
+      this.trackTarget.addEventListener("componentremoved", e => {
```

Comment:
Is there a better way to do this? Trying to detect removal of the element.

Reply:
The "correct" way is probably using a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) but in this case I would probably just check for `this.trackTarget && !this.trackTarget.parrentNode` in `tick`.